### PR TITLE
UI Element Decriber Integration

### PR DESCRIPTION
**Pull request by BrandonMagana**
This PR adds the following:

- Element Describer:  Interface and implementation for describing a ui element based  on an  b64 encoded image
- Image Cropper: generates crops of and image
- Image Encoder.  encodes images in base 64, this enable us to forward them to the Element Describer.
-  Support for env variables using python-dotenv

### CODE REVIEW

File: server/api.py [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
 from adapters.object_detection import YOLO
+from adapters.element_describer import ChatGPTDescriber
 from fastapi import FastAPI, File, UploadFile
+from utils.image_cropper import ImageCropper
+from utils.image_encoder import b64_encode_image
 from PIL import Image
 
 app = FastAPI()
 od_model = YOLO()
+describer = ChatGPTDescriber()
```
Comment (CarlosCRG19) [inline server/api.py:12]: api key and project are missing, which I think will be introduced together with the `dotenv` package

File: server/api.py [hunk 2]
```diff
@@ -26,6 +30,14 @@ async def detect_objects(file: UploadFile = File(...)) -> dict:
 
     objects = od_model.detect_objects(image)
 
+    cropper = ImageCropper(image=image)
+
+    descriptions = []
+    for crop in cropper.get_image_crops(objects):
+        b64_crop = b64_encode_image(crop)
+        description = describer.describe(b64_crop)
+        descriptions.append(description)
```
Comment (CarlosCRG19) [inline server/api.py:39]: nice!

File: server/adapters/element_describer.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
+from abc import ABC, abstractmethod
```
Comment (CarlosCRG19) [inline server/adapters/element_describer.py:1]: same comments as in #4

File: server/setup.sh [hunk 4]
```diff
@@ -0,0 +1,5 @@
+# Bash script to setup the environment for the project
+python3 -m venv env
+source ./env/bin/activate
+pip install -r requirements.txt
+
```
Comment (CarlosCRG19) [inline server/setup.sh:5]: this file can be removed with the introduction of #6 but it is okay to merge it with this PR and then erase it

File: server/utils/image_cropper.py [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+class ImageCropper:
+    def __init__(self, image: Image.Image):
+        self.image = image
+
+    def get_image_crops(self, detected_objects: list[DetectedObject]) -> list[Image.Image]:
+        cropped_elements = []
+        for detected_object in detected_objects:
+            bbox = detected_object.bbox
+            crop = (bbox.x1, bbox.y1, bbox.x2, bbox.y2)
+            cropped_element = self.image.crop(crop)
```
Comment (CarlosCRG19) [inline server/utils/image_cropper.py:14]: `crop` method receives a tuple of ints according to docs: https://pillow.readthedocs.io/en/stable/reference/Image.html#PIL.Image.Image.crop

But here it is receiving a tuple of ints...
```suggestion
            crop = (int(bbox.x1), int(bbox.y1), int(bbox.x2), int(bbox.y2))
            cropped_element = self.image.crop(crop)
```

File: server/api.py (hunk 1 above)
Comment (BrandonMagana) [reply server/api.py:12]: I think that is added on #4 , @antonieto  already has a commit for that

File: server/api.py (hunk 2 above)
Comment (BrandonMagana) [reply server/api.py:39]: thanks!

File: server/adapters/element_describer.py (hunk 3 above)
Comment (BrandonMagana) [reply server/adapters/element_describer.py:1]: You  think I should integrate those changes here? Or should @antonieto fix them?

File: server/setup.sh (hunk 4 above)
Comment (BrandonMagana) [reply server/setup.sh:5]: lets remove it here

File: server/utils/image_cropper.py (hunk 5 above)
Comment (BrandonMagana) [reply server/utils/image_cropper.py:14]: I just changed it. Thanks for the suggestion.

File: server/api.py (hunk 1 above)
Comment (BrandonMagana) [reply server/api.py]: Done, check it out

**Review by CarlosCRG19 — APPROVED** [review]

**Review by antonieto — COMMENTED** [review]
nice!

File: server/adapters/element_describer.py (hunk 3 above)
Comment (antonieto) [reply server/adapters/element_describer.py:1]: i addressed these comments @CarlosCRG19

**Pull request merged**